### PR TITLE
Refactor CloudFerry utils

### PR DESCRIPTION
**Pull request by vladiskuz**
Clean up cloudferrylib/utils/utils.py file. Format code according the pep8 and the OpenStack Style Guidelines.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.utils.utils
cloudferrylib/utils/utils.py:124: [W0201(attribute-defined-outside-init), Postman.__enter__] Attribute 'server' defined outside __init__
cloudferrylib/utils/utils.py:145: [W0232(no-init), Templater] Class has no __init__ method
cloudferrylib/utils/utils.py:165: [W0613(unused-argument), get_log] Unused argument 'name'
cloudferrylib/utils/utils.py:204: [W0201(attribute-defined-outside-init), forward_agent.__enter__] Attribute 'pid' defined outside __init__
cloudferrylib/utils/utils.py:205: [W0201(attribute-defined-outside-init), forward_agent.__enter__] Attribute 'ssh_auth_sock' defined outside __init__
cloudferrylib/utils/utils.py:264: [W0201(attribute-defined-outside-init), UpSshTunnelClass.__enter__] Attribute 'port' defined outside __init__

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1799/

File: cfglib.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from oslo.config import cfg
+from pkg_resources import Requirement, resource_filename
```
Comment (SVilgelm) [inline cfglib.py]: Please, Do not import objects, only modules.

**Comment by SVilgelm** [comment]
You made lots of changes on utils.py file, Could you also rewrite according OpenStack Code Guidelines?

**Comment by roman-verchikov** [comment]
Please resolve unit tests failures

**Comment by abochkarev** [comment]
There is a suggestion to move a logging related functionality from utils.py to other file (for example: logging/log.py).  
PS: I know that lots of imports will be changed...

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloud.cloud
cloud/cloud.py:22: [W0404(reimported), ] Reimport 'utils' (imported line 17)
************* Module cloudferrylib.os.actions.convert_file
cloudferrylib/os/actions/convert_file.py:17: [W0612(unused-variable), ConvertFile.run] Unused variable 'instance'
************* Module cloudferrylib.os.actions.utils
cloudferrylib/os/actions/utils.py:41: [E1102(not-callable), transfer_file_to_file] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.os.actions.transport_ephemeral
cloudferrylib/os/actions/transport_ephemeral.py:175: [W0612(unused-variable), TransportEphemeral.copy_ephemeral_iscsi_to_ceph] Unused variable 'inst_id'
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_ceph
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:29: [W0221(arguments-differ), SSHCephToCeph.transfer] Arguments number differs from overridden 'transfer' method
cloudferrylib/utils/drivers/ssh_ceph_to_ceph.py:34: [E1129(not-context-manager), SSHCephToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_file_to_ceph
cloudferrylib/utils/drivers/ssh_file_to_ceph.py:35: [E1129(not-context-manager), SSHFileToCeph.transfer] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module cloudferrylib.utils.drivers.ssh_ceph_to_file
cloudferrylib/utils/drivers/ssh_ceph_to_file.py:31: [E1102(not-callable), SSHCephToFile.transfer] utils.up_ssh_tunnel is not callable
************* Module cloudferrylib.os.network.nova_network
cloudferrylib/os/network/nova_network.py:58: [W0221(arguments-differ), NovaNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/nova_network.py:61: [W0221(arguments-differ), NovaNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
************* Module cloudferrylib.utils.ssh_util
cloudferrylib/utils/ssh_util.py:17: [W0403(relative-import), ] Relative import 'cmd_cfg', should be 'cloudferrylib.utils.cmd_cfg'
cloudferrylib/utils/ssh_util.py:18: [W0403(relative-import), ] Relative import 'utils', should be 'cloudferrylib.utils.utils'
************* Module cloudferrylib.utils.drivers.ssh_file_to_file
cloudferrylib/utils/drivers/ssh_file_to_file.py:35: [E1102(not-callable), SSHFileToFile.transfer] utils.up_ssh_tunnel is not callable
cloudferrylib/utils/drivers/ssh_file_to_file.py:79: [E1129(not-context-manager), SSHFileToFile.transfer_direct] Context manager 'tuple' doesn't implement __enter__ and __exit__.
************* Module tests.cloud.test_grouping
tests/cloud/test_grouping.py:40: [E1002(super-on-old-class), GroupingTestCase.setUp] Use of super on an old style class
tests/cloud/test_grouping.py:96: [E1002(super-on-old-class), GroupingTestCase.tearDown] Use of super on an old style class
************* Module tests.cloudferrylib.os.actions.test_converter_volume_to_image
tests/cloudferrylib/os/actions/test_converter_volume_to_image.py:25: [E1002(super-on-old-class), ConverterVolumeToImageTest.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:232: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:312: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:376: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
************* Module tests.cloudferrylib.os.actions.test_copy_g2g
tests/cloudferrylib/os/actions/test_copy_g2g.py:25: [E1002(super-on-old-class), CopyFromGlanceToGlanceTestCase.setUp] Use of super on an old style class
************* Module tests.cloudferrylib.os.actions.test_get_info_images
tests/cloudferrylib/os/actions/test_get_info_images.py:25: [E1002(super-on-old-class), GetInfoImagesTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.utils.utils
cloudferrylib/utils/utils.py:140: [W0201(attribute-defined-outside-init), Postman.__enter__] Attribute 'server' defined outside __init__
cloudferrylib/utils/utils.py:181: [W0613(unused-argument), get_log] Unused argument 'name'
cloudferrylib/utils/utils.py:219: [W0201(attribute-defined-outside-init), ForwardAgent.__enter__] Attribute 'pid' defined outside __init__
cloudferrylib/utils/utils.py:220: [W0201(attribute-defined-outside-init), ForwardAgent.__enter__] Attribute 'ssh_auth_sock' defined outside __init__
cloudferrylib/utils/utils.py:278: [W0201(attribute-defined-outside-init), UpSshTunnelClass.__enter__] Attribute 'port' defined outside __init__
************* Module tests.cloudferrylib.os.compute.test_server_groups
tests/cloudferrylib/os/compute/test_server_groups.py:45: [E1002(super-on-old-class), ServerGroupTestCase.setUp] Use of super on an old style class
tests/cloudferrylib/os/compute/test_server_groups.py:109: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists] Access to a protected member _delete_server_group of a client class
tests/cloudferrylib/os/compute/test_server_groups.py:110: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists] Access to a protected member _deploy_server_group of a client class
tests/cloudferrylib/os/compute/test_server_groups.py:113: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists] Access to a protected member _delete_server_group of a client class
tests/cloudferrylib/os/compute/test_server_groups.py:114: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists] Access to a protected member _deploy_server_group of a client class
tests/cloudferrylib/os/compute/test_server_groups.py:129: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists_doesnt_match] Access to a protected member _delete_server_group of a client class
tests/cloudferrylib/os/compute/test_server_groups.py:130: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists_doesnt_match] Access to a protected member _deploy_server_group of a client class
tests/cloudferrylib/os/compute/test_server_groups.py:133: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists_doesnt_match] Access to a protected member _delete_server_group of a client class
tests/cloudferrylib/os/compute/test_server_groups.py:140: [W0212(protected-access), ServerGroupTestCase.test_deploy_server_groups_already_exists_doesnt_match] Access to a protected member _deploy_server_group of a client class
************* Module tests.cloudferrylib.os.compute.test_cold_evacuate
tests/cloudferrylib/os/compute/test_cold_evacuate.py:14: [I0011(locally-disabled), ] Locally disabling no-member (E1101)
************* Module tests.cloudferrylib.os.image.test_glance_image
tests/cloudferrylib/os/image/test_glance_image.py:46: [E1002(super-on-old-class), GlanceImageTestCase.setUp] Use of super on an old style class
tests/cloudferrylib/os/image/test_glance_image.py:50: [W0212(protected-access), GlanceImageTestCase.setUp] Access to a protected member _resp of a client class
************* Module tests.cloudferrylib.os.network.test_nova_network
tests/cloudferrylib/os/network/test_nova_network.py:37: [E1002(super-on-old-class), TestNovaNetwork.setUp] Use of super on an old style class
************* Module tests.cloudferrylib.os.compute.test_nova
tests/cloudferrylib/os/compute/test_nova.py:44: [E1002(super-on-old-class), NovaComputeTestCase.setUp] Use of super on an old style class
tests/cloudferrylib/os/compute/test_nova.py:338: [W0212(protected-access), FlavorDeploymentTestCase.test_flavor_is_updated_with_destination_id] Access to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:339: [W0212(protected-access), FlavorDeploymentTestCase.test_flavor_is_updated_with_destination_id] Access to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:365: [W0212(protected-access), FlavorDeploymentTestCase.test_flavor_is_updated_with_destination_id] Access to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:392: [W0212(protected-access), FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:395: [W0212(protected-access), FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:397: [W0212(protected-access), FlavorDeploymentTestCase.test_flavor_is_not_created_if_already_exists_on_dest] Access to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:416: [W0212(protected-access), FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:417: [W0212(protected-access), FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:421: [W0212(protected-access), FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:423: [W0212(protected-access), FlavorDeploymentTestCase.test_access_not_updated_for_public_flavors] Access to a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:442: [W0212(protected-access), FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to a protected member _create_flavor_if_not_exists of a client class
tests/cloudferrylib/os/compute/test_nova.py:443: [W0212(protected-access), FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to a protected member _add_flavor_access_for_tenants of a client class
tests/cloudferrylib/os/compute/test_nova.py:447: [W0212(protected-access), FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to a protected member _deploy_flavors of a client class
tests/cloudferrylib/os/compute/test_nova.py:449: [W0212(protected-access), FlavorDeploymentTestCase.test_access_list_is_updated_for_non_public_flavors] Access to a protected member _add_flavor_access_for_tenants of a client class
************* Module tests.cloudferrylib.os.identity.test_keystone
tests/cloudferrylib/os/identity/test_keystone.py:49: [E1002(super-on-old-class), KeystoneIdentityTestCase.setUp] Use of super on an old style class
tests/cloudferrylib/os/identity/test_keystone.py:214: [W0212(protected-access), KeystoneIdentityTestCase.test_read_info] Access to a protected member _get_user_roles_cached of a client class
tests/cloudferrylib/os/identity/test_keystone.py:215: [W0212(protected-access), KeystoneIdentityTestCase.test_read_info] Access to a protected member _get_user_roles_cached of a client class
tests/cloudferrylib/os/identity/test_keystone.py:237: [W0613(unused-argument), KeystoneIdentityTestCase.test_deploy.tenant_create] Unused argument 'kwargs'
tests/cloudferrylib/os/identity/test_keystone.py:242: [W0613(unused-argument), KeystoneIdentityTestCase.test_deploy.user_create] Unused argument 'kwars'
tests/cloudferrylib/os/identity/test_keystone.py:247: [W0613(unused-argument), KeystoneIdentityTestCase.test_deploy.roles_create] Unused argument 'role_name'
tests/cloudferrylib/os/identity/test_keystone.py:447: [W0212(protected-access), KeystoneClientTestCase.test_adds_region_if_set_in_config] Access to a protected member _get_client_by_creds of a client class
tests/cloudferrylib/os/identity/test_keystone.py:479: [W0212(protected-access), KeystoneClientTestCase.test_does_not_add_region_if_not_set_in_config] Access to a protected member _get_client_by_creds of a client class
************* Module tests.cloudferrylib.os.storage.test_cinder_netapp
tests/cloudferrylib/os/storage/test_cinder_netapp.py:48: [E1002(super-on-old-class), CinderNetAppTestCase.setUp] Use of super on an old style class
************* Module tests.cloudferrylib.utils.test_runner
tests/cloudferrylib/utils/test_runner.py:41: [W0703(broad-except), RemoteRunnerTestCase.test_errors_are_suppressed_for_run_ignoring_errors] Catching too general exception Exception
************* Module tests.cloudferrylib.utils.test_utils
tests/cloudferrylib/utils/test_utils.py:33: [W0212(protected-access), ForwardAgentTestCase.test_agent_is_already_run_w_keys] Access to a protected member _agent_already_running of a client class
tests/cloudferrylib/utils/test_utils.py:43: [W0212(protected-access), ForwardAgentTestCase.test_agent_is_not_run] Access to a protected member _agent_already_running of a client class
tests/cloudferrylib/utils/test_utils.py:53: [W0212(protected-access), ForwardAgentTestCase.test_agent_is_already_run_w_another_key] Access to a protected member _agent_already_running of a client class
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module tests.cloudferrylib.os.storage.test_cinder_storage
tests/cloudferrylib/os/storage/test_cinder_storage.py:46: [E1002(super-on-old-class), CinderStorageTestCase.setUp] Use of super on an old style class
tests/cloudferrylib/os/storage/test_cinder_storage.py:232: [W0703(broad-except), CinderStorageTestCase.test_get_volume_path_iscsi_error] Catching too general exception Exception

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
Yeaaaaah, you did it. So, you touched lots of files and got bunch of pylint issues, please, resolve all from the above list :) I believe, you can do it.

**Comment by vladiskuz** [comment]
I fixed utils.py module, but it affected the others files. I have no so much time to fix all 35 files. This pull request needed for SAIC-186.
I guess that pylint need adjustment because it has very strange rules right now.

**Comment by vladiskuz** [comment]
Move to https://github.com/MirantisWorkloadMobility/CloudFerry/pull/703

**Pull request closed without merging**